src/client/lwm2m_coap_client.erl

```diff
@@ -100,7 +108,8 @@ split_path([$/]) -> [];
 split_path([$/ | Path]) -> split_segments(Path, $/, []).
 
 split_query([]) -> [];
-split_query([$? | Path]) -> split_segments(Path, $&, []).
+split_query([$? | Path]) -> split_segments(Path, $&, []);
+split_query(Path) -> split_segments(Path, $&, []).
```

Comment:
dead-loop ?

Reply:
won't be a dead-loop here, `split_segments/3` won't call `split_query/1` back.
the leading `?` in path is removed after switch to `emqx_http_lib:uri_parse/1`, we need to handle it correctly